lib/omnibus/util.rb

```diff
@@ -220,5 +290,24 @@ def create_link(a, b)
       log.debug(log_key) { "Linking `#{a}' to `#{b}'" }
       FileUtils.ln_s(a, b)
     end
+
+    private
+
+    def filter_paths
+      @filter_paths ||=
+        begin
+          # Any alternate binaries or gems from the users environment.
+          filters = Gem.paths.path.dup
+          # Any possible embedded ruby in chef products.
+          if windows?
+            filters << "C:/opscode"
+          else
+            filters << "/opt/chef" << "/opt/chefdk" << "/opt/delivery-cli"
```

Comment:
That's fine but we should not be playing whack-a-mole with every bad thing that could end up in the `PATH`. That is a losing battle which we'll never cover every case for. Perhaps you are better off building up a fresh/clean `PATH` for the affected functions/commands from `Omnibus::Builder`. It would be preferable to add `C:\msys2\bin`, `C:\msys2\mingw32\bin`, and `C:\msys2\usr\bin` to a clean `PATH` (assuming they exist). We should also isolate this special behavior to Windows only...no reason to muck with *nix because of Windows/MSYS2 sins.